source/common/grpc/common.cc

```diff
@@ -56,8 +56,8 @@ absl::optional<Status::GrpcStatus> Common::getGrpcStatus(const Http::HeaderMap&
     return absl::nullopt;
   }
   if (!absl::SimpleAtoi(grpc_status_header->value().getStringView(), &grpc_status_code) ||
-      grpc_status_code > Status::GrpcStatus::MaximumValid) {
-    return {Status::GrpcStatus::InvalidCode};
+      grpc_status_code > Status::WellKnownGrpcStatus::MaximumKnown) {
```

Comment:
I think you want to allow the condition `grpc_status_code > Status::WellKnownGrpcStatus::MaximumKnown`? Though we might want to guard this with an additional parameter. Make this method `getGrpcStatus(const Http::HeaderMap& trailers, bool allow_user_defined = false)`?

Reply:
I think if we applied `allow_user_defined` param, user must add management flag of this in grpc-json transcoder config. Is it ok?

Reply:
I think if we applied allow_user_defined param, user must add management flag of this in grpc-json transcoder config. Is it ok?

Reply:
In grpc-json-transcoder you should always set it to `true`, for other places keep it `false` should be fine.